Kinvey/KinveyTests/MultiInsertSpec.swift

```diff
@@ -0,0 +1,1659 @@
+                        default:
+                            return HttpResponse(request: request, urlProcotolType: KinveyURLProtocol.self)
+                        }
+                    }
+                    defer {
+                        setURLProtocol(nil)
+                    }
+                    
+                    print((syncDataStore.sync?.sync as? RealmSync<Person>)?.realm.configuration.fileURL?.path)
+                    
+                    let result = kinveySaveMulti(
+                        dataStore: syncDataStore,
+                        entities: [
+                            Person { $0.geolocation = GeoPoint(latitude: -300, longitude: -300) },
+                            Person { $0.geolocation = GeoPoint(latitude: -300, longitude: -300) }
+                        ]
+                    ).result
+                    expect(result).toNot(beNil())
+                    guard let resultUnwrapped = result else {
+                        return
+                    }
+                    expect(resultUnwrapped.entities.count).to(equal(2))
+                    expect(resultUnwrapped.errors.count).to(equal(0))
+                
+                    expect(syncDataStore.pendingSyncCount()).to(equal(2))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(2))
+                    expect(syncDataStore.pendingSyncOperations().count).to(equal(1))
+                    
+                    let errors = kinveySync(dataStore: syncDataStore).errors
+                    expect(errors?.count).to(equal(2))
+                    expect(errors?.first?.localizedDescription).to(equal(errorMessage))
+                    expect(errors?.last?.localizedDescription).to(equal(errorMessage))
+                    
+                    let multiSaveError1 = errors?.first as? MultiSaveError
+                    let multiSaveError2 = errors?.last as? MultiSaveError
+                    
+                    expect(multiSaveError1).toNot(beNil())
+                    expect(multiSaveError2).toNot(beNil())
+                    
+                    expect(multiSaveError1?.index).to(equal(0))
+                    expect(multiSaveError2?.index).to(equal(1))
+                    
+                    expect(multiSaveError1?.code).to(equal(errorCode))
+                    expect(multiSaveError2?.code).to(equal(errorCode))
+                    
+                    expect(multiSaveError1?.message).to(equal(errorMessage))
+                    expect(multiSaveError2?.message).to(equal(errorMessage))
+                    
+                    expect(syncDataStore.pendingSyncCount()).to(equal(2))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(2))
+                    expect(syncDataStore.pendingSyncOperations().count).to(equal(1))
+                
+                    let entities = kinveyFind(dataStore: syncDataStore).entities
+                    expect(entities?.count).to(equal(0))
```

Comment:
@Martin-Hristov I had this previously expecting to return `2`, but then I figured... if the insert failed, it will never work, even if we try again. What do you think? Do you think in this case we should remove the entities from the local cache?

Reply:
fixed